### PR TITLE
[ONNX] Fix export for tuple outputs

### PR DESCRIPTION
**Pull request by KsenijaS**
Fix export for tuple outputs in tracing.

### CODE REVIEW

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -465,7 +465,10 @@ def _model_to_graph(model, args, verbose=False,
         if not (isinstance(torch_out, list) or isinstance(torch_out, tuple)):
             output_wrapped = [torch_out]  # type: ignore
         else:
-            output_wrapped = torch_out  # type: ignore
+            output_wrapped = []  # type: ignore
+            for el in torch_out:
+                if el is not None:
+                    output_wrapped.append(el)
```
Comment (BowenBao) [inline torch/onnx/utils.py:471]: - does it support None in tuple/list/nested?
- will this affect scripting?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -1532,6 +1532,23 @@ def forward(self, x):
                       dynamic_axes={'input_1': [0, 1, 2],
                                     'output_1': [0, 1, 2]})
 
+    @disableScriptTest()
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1535]: If you are disbling scripting, why are you testing with if/else? This is not reflected in the model in tracing.

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -468,7 +481,7 @@ def _model_to_graph(model, args, verbose=False,
         if not (isinstance(torch_out, list) or isinstance(torch_out, tuple)):
             output_wrapped = [torch_out]  # type: ignore
         else:
-            output_wrapped = torch_out  # type: ignore
+            output_wrapped = _remove_none_list(torch_out)  # type: ignore
 
         output_tensors, out_desc = torch._C._jit_flatten(tuple(output_wrapped))
         torch._C._jit_pass_onnx_assign_output_shape(graph, output_tensors, out_desc, _onnx_shape_inference)
```
Comment (BowenBao) [inline torch/onnx/utils.py:487]: the `out_desc` afterwards is different, is that covered?

This doesn't feel compatible when support is added for None output from script module. Is there a design to be extensible for that?

**Review by shubhambhokare1 — COMMENTED** [review]
Seems like PR #54623 is similar to this PR but handles ScriptModel cases as well

**Comment by KsenijaS** [comment]
> Seems like PR #54623 is similar to this PR but handles ScriptModel cases as well
that's a good point. I can close my PR then. Although I am not sure if we should remove None from tuple outputs in scripting.

**Pull request closed without merging**